CONTRIBUTING.md

```diff
@@ -0,0 +1,113 @@
+* `ghcr.io/bufbuild/plugins-<org>-<name>:<version>-buildcache` (Registry cache with `mode=max`)
+
+These images are used only for caching - the authoritative images used for plugin execution are pushed to the BSR.
+Untagged versions of these cached images can be cleaned up at any time (only the latest tagged versions are used).
+
+During development, we've encountered a number of issues which lead to cache misses and in turn cause unnecessary revisions of plugins to be published to the BSR.
+Workarounds are in place for the majority of issues, however we should be careful to ensure build reproducibility.
+One critical change to the build images is the removal of `--build-arg` arguments in base images and the migration from `type=inline` to `type=registry` caching.
+See [PR #41](https://github.com/bufbuild/plugins/pull/41) and [PR #38](https://github.com/bufbuild/plugins/pull/38) for the gory details.
+
+## Updates
+
+### Creating a new plugin version
+
+If the `fetcher` command opens a PR for a new version of an existing plugin, there are currently some manual steps required before merging:
+
+1. Check out the PR locally (i.e. `gh pr checkout <num>`).
+2. Perform any manual updates to `package.json`, `go.mod`, or other files (if used by the plugin).
+   If NPM or Go dependencies change, ensure that `npm i` or `go mod tidy` are run to ensure that the `package-lock.json` or `go.sum` files are kept up to date.
+3. Double check version numbers in `plugin.yaml` or other files to ensure they are bumped.
+   Often times, there may be dependencies (plugin or runtime dependencies) which need to be updated.
+4. Run `make` and `make test` to build the plugin and run the unit tests.
+   It is expected that the tests will fail with a directory checksum mismatch (because the `plugin.sum` file doesn't exist).
+   Perform a manual inspection of the generated code for the new version of the plugin under `tests/testdata/library/<your-plugin>/v<version>/*/gen` to ensure it looks correct.
+   It may be helpful to diff a previous version of the plugin's generated code to the new version's to verify expected changes are included.
+5. Add the `plugin.sum` files for the new version of the generated code to source control and request a PR review for any updates.
+
+### Updating Docker Base Images
+
+We've only included a representative set of Docker images in the [.github/dependabot.yml](https://github.com/bufbuild/plugins/blob/main/.github/dependabot.yml) file for each type of plugin (Node, Go, etc.).
+If Dependabot detects an update to a base image, we should update all plugins to use the new base image either with a separate PR or a manual update to the Dependabot PR.
+
+## Local Testing
+
+When testing locally, you may wish to build for a different architecture or push plugins to a different instance of the BSR.
+For example:
+
+```
+$ make push DOCKER_ORG="bufbuild.internal" BUF_PLUGIN_PUSH_ARGS="--override-remote bufbuild.internal"
```

Comment:
Still working to get this working locally. I'd like to use consistent ways to build in both CI and locally. Right now we're still using `buf alpha plugin push` support for building Docker images locally.

Reply:
Tbh I wouldn't worry about this too much, and if it adds complexity I'd say the tradeoff isn't worth it

We rebuild clusters and prune images quite often, so building/pushing all these plugins (which will grow over time) to the local cluster isn't something folks will be doing because it decreases the development feedback loop.

I think it's enough to be able to clone this repository, and with `--override-remote` target the local cluster to upload specific plugin(s).